src/dashboard/src/components/administration/views.py

```diff
@@ -386,9 +386,18 @@ def _usage_get_directory_used_bytes(path):
     except OSError:
         logger.exception("No such directory: %s", path)
         return 0
-    except subprocess.CalledProcessError:
-        logger.exception("Unable to determine usage of %s.", path)
-        return 0
+    except subprocess.CalledProcessError as err:
+        # CalledProcessErrors are typically the result of du returning
+        # 1 because there were some directories in the path for which
+        # the archivematica user doesn't have permissions. In this case
+        # du still prints an output to stdout so we try to catch it
+        # from CalledProcessError.output.
+        logger.warning(
+            "Non-zero exit code while determining usage of %s. Some directories may be missing from total.",
+            path,
+        )
+        byte_count = err.output.split("\t")[0]
+        return byte_count if byte_count else 0
```

Comment:
Two things that we should probably consider:
* `err.output` can be `None`
* Using `int(byte_count)` (raises `ValueError`) would ensure that we're returning an integer

Reply:
Thanks @sevein ! https://github.com/artefactual/archivematica/pull/1640/commits/860508a9616edd1ba08d9dd13feb3abaa382d5db refactors this to return 0 for an `AttributeError` (if `err.output` is `None`) or `ValueError` (if `err.output` can't be cast to an `int`), and otherwise return the number of bytes as an `int`.